app/assets/javascripts/application.js.erb

```diff
@@ -205,35 +205,50 @@ var _Lobsters = Class.extend({
       action, { reason: reason });
   },
 
-  postComment: function(form) {
+  postComment: function(btn, form) {
     var params = $(form).serializeArray();
     params.push({"name": "show_tree_lines", "value": "true"});
 
-    $.post($(form).attr("action"), params, function(data) {
-      // Clear form: Firefox will keep form values on reload (e.g. F5), which isn't too useful if
-      // it's already posted.
-      $(form).find('textarea').val('')
-
-      var parsedHTML = $.parseHTML(data);
-
-      var isEdit = $(form).hasClass("edit_comment");
-      var hasChildren = $(form).closest(".comments_subtree")
-        .find(".comments_subtree").length;
-      if (isEdit && hasChildren) {
-        $(parsedHTML).find(".comment_parent_tree_line:first")
-          .removeClass("no_children");
-      }
+    var done = Lobsters.loading(btn, () => {
+      $.post($(form).attr("action"), params, function(data) {
+        // Clear form: Firefox will keep form values on reload (e.g. F5), which isn't too useful if
+        // it's already posted.
+        $(form).find('textarea').val('')
 
-      if ($(form).find("#parent_comment_short_id").length) {
-        // reply to comment
-        $(form).closest(".comments_subtree")
-          .find(".comment_parent_tree_line:first").removeClass("no_children");
-        $(form).closest(".comment").replaceWith(parsedHTML);
-      } else {
-        // reply to story
-        $(form).parent(".comment").replaceWith(parsedHTML);
-      }
-    });
+        var parsedHTML = $.parseHTML(data);
+
+        var isEdit = $(form).hasClass("edit_comment");
+        var hasChildren = $(form).closest(".comments_subtree")
+          .find(".comments_subtree").length;
+        if (isEdit && hasChildren) {
+          $(parsedHTML).find(".comment_parent_tree_line:first")
+            .removeClass("no_children");
+        }
+
+        if ($(form).find("#parent_comment_short_id").length) {
+          // reply to comment
+          $(form).closest(".comments_subtree")
+            .find(".comment_parent_tree_line:first").removeClass("no_children");
+          $(form).closest(".comment").replaceWith(parsedHTML);
+        } else {
+          // reply to story
+          $(form).parent(".comment").replaceWith(parsedHTML);
+        }
+      })
+        .always(() => done())
+        .fail(function(req, err) { alert(`could not post comment: ${err}`) })
```

Comment:
Rather than add a modal `alert`, please add like `<div class="error"></div>` to the form and dump the error in there. And have the `submit` button clear that.